pkg/feature/api/validation.go

```diff
@@ -1154,7 +1158,36 @@ func validateRule(variations []*featureproto.Variation, rule *featureproto.Rule,
 		}
 		return dt.Err()
 	}
-	return validateStrategy(variations, rule.Strategy, localizer)
+	// Check depencendy.
+	tarF.Rules = append(tarF.Rules, rule)
+	defer func() { tarF.Rules = tarF.Rules[:len(tarF.Rules)-1] }()
+	err := validateFeatureDependencies(fs)
+	if err != nil {
+		if err == evaluation.ErrCycleExists {
```

Comment:
The `errors.Is` is preferable to `==` operator.
```suggestion
		if errors.Is(err, evaluation.ErrCycleExists) {
```